internal/store/backup_metadata.go

```diff
@@ -0,0 +1,289 @@
+package store
+
+import (
+	"database/sql"
+	"encoding/json"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+)
+
+const (
+	backupMetadataTable = "BackupMetadata"
+)
+
+var backupMetadataSelect sq.SelectBuilder
+
+func init() {
+	backupMetadataSelect = sq.
+		Select(
+			"ID", "InstallationID", "ClusterInstallationID", "DataResidenceRaw", "State", "RequestAt", "StartAt", "DeleteAt", "APISecurityLock", "LockAcquiredBy", "LockAcquiredAt",
+		).
+		From(backupMetadataTable)
+}
+
+type rawBackupMetadata struct {
+	*model.BackupMetadata
+	DataResidenceRaw []byte
+}
+
+type rawBackupsMetadata []*rawBackupMetadata
+
+func (r *rawBackupMetadata) toBackupMetadata() (*model.BackupMetadata, error) {
+	// We only need to set values that are converted from a raw database format.
+	var err error
+	if len(r.DataResidenceRaw) > 0 {
+		dataResidence := model.S3DataResidence{}
+		err = json.Unmarshal(r.DataResidenceRaw, &dataResidence)
+		if err != nil {
+			return nil, err
+		}
+		r.BackupMetadata.DataResidence = &dataResidence
+	}
+
+	return r.BackupMetadata, nil
+}
+
+func (r *rawBackupsMetadata) toBackupsMetadata() ([]*model.BackupMetadata, error) {
+	if r == nil {
+		return []*model.BackupMetadata{}, nil
+	}
+	backupsMeta := make([]*model.BackupMetadata, 0, len(*r))
+
+	for _, raw := range *r {
+		metadata, err := raw.toBackupMetadata()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create backup metadata from raw")
+		}
+		backupsMeta = append(backupsMeta, metadata)
+	}
+	return backupsMeta, nil
+}
+
+// IsBackupRunning checks if any backup is currently running or requested for specified installation.
+func (sqlStore *SQLStore) IsBackupRunning(installationID string) (bool, error) {
+	var totalResult countResult
+	builder := sq.
+		Select("Count (*)").
+		From(backupMetadataTable).
+		Where("InstallationID = ?", installationID).
+		Where(sq.Eq{"State": model.AllBackupMetadataStatesPendingWork}).
```

Comment:
Just out of curiosity, what's the difference between this `sq.Eq` struct being passed to `Where(` and the more-common `thing = ?` statement like the one on the previous line?

Reply:
Good question!
In this case we want the `State` be in one of states from `model.AllBackupMetadataStatesPendingWork` as this is a slice.

Using `Where("State = ?", model.AllBackupMetadataStatesPendingWork)` would evaluate simply to `... AND State = ? ...` and would not work because we pass slice. Using `sq.Eq` on the other hand will result in query like that `...  AND State IN (?,?) ...` which is what we are shooting for.